src/AspNetCore/Src/forwardinghandler.cxx

```diff
@@ -1534,7 +1529,36 @@ REQUEST_NOTIFICATION_STATUS
             reinterpret_cast<PVOID>(static_cast<DWORD_PTR>(hrCompletionStatus)));
     }
 
-    if (m_pApplication->QueryConfig()->QueryHostingModel() == HOSTING_OUT_PROCESS)
+    // Application is null here? Can this happen for out of proc scenario?
+    if (m_pApplication->QueryConfig()->QueryHostingModel() == HOSTING_IN_PROCESS)
+    {
+        if (FAILED(hrCompletionStatus))
+        {
+            return RQ_NOTIFICATION_FINISH_REQUEST;
+        }
+        else
+        {
+            // For now wea re assuming we are in our own self contained box. 
+            // TODO refactor Finished and Failure sections to handle in process and out of process failure.
+            IN_PROCESS_APPLICATION* application = (IN_PROCESS_APPLICATION*)m_pApplication;
+            if (application == NULL)
+            {
+                hr = E_FAIL;
+                return RQ_NOTIFICATION_FINISH_REQUEST;
+            }
+
+            if (application->QueryIsWebSocketsConnection())
+            {
+                // If this event occurs, we need to let the registered callback handler take care of this.
+                return RQ_NOTIFICATION_PENDING;
```

Comment:
this will lead to leak of request as no one will return continue

Reply:
This is a safety check if the websocket connection causes an OnAsyncCompletion event to be triggered. Continue would eventually be called once the websocket connection is closed.